### PR TITLE
[BUG] [Java] Add Generated annotations to more classes 

### LINKED ISSUE #17735: [BUG] [JAVA] Missing Generated annotations on classes
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description
Some classes for the native java client does not have generated annotations, and are therefore flagged by tools like jacoco.

This includes
* RFC3339DateFormat
* ServerConfiguration
* ServerVariable
* ApiResponse
* ClassDiscriminatorMapping in `JSON.mustache`
* TypeReference<T> in `api.mustache`

##### openapi-generator version
Using 7.1.0 but the problem is also present on master.

##### OpenAPI declaration file content or url
```json
{
  "openapi": "3.0.3",
  "info": {
    "version": "1.0.0",
    "description": "Spec to show missing Generated annotations",
    "title": "Minimal example"
  },
  "paths": {
    "/whatever": {
      "get": {
        "operationId": "operation",
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "field": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "model1": {
        "type": "object",
        "properties": {
          "field": {
            "type": "string"
          }
        }
      }
    }
  }
}

```

##### Generation Details
Generated using `mvn compile`.

##### Steps to reproduce
Using this `pom.xml`:
```xml
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.example</groupId>
  <artifactId>openapi-missing-generated-annotations</artifactId>
  <packaging>jar</packaging>
  <version>1.0.0</version>

  <dependencies>
    <dependency>
      <groupId>com.fasterxml.jackson.core</groupId>
      <artifactId>jackson-databind</artifactId>
      <version>2.16.1</version>
    </dependency>
    <dependency>
      <groupId>com.fasterxml.jackson.datatype</groupId>
      <artifactId>jackson-datatype-jsr310</artifactId>
      <version>2.16.1</version>
    </dependency>
    <dependency>
      <groupId>jakarta.annotation</groupId>
      <artifactId>jakarta.annotation-api</artifactId>
      <version>2.1.1</version>
    </dependency>
  </dependencies>

  <build>
    <plugins>
      <plugin>
        <groupId>org.openapitools</groupId>
        <artifactId>openapi-generator-maven-plugin</artifactId>
        <version>7.2.0</version>
        <executions>
          <execution>
            <goals>
              <goal>generate</goal>
            </goals>
            <configuration>
              <apiNameSuffix>Client</apiNameSuffix>
              <inputSpec>${project.basedir}/openapi.json</inputSpec>
              <generatorName>java</generatorName>

              <apiPackage>com.example.rest.client.api</apiPackage>
              <generateApiTests>false</generateApiTests>
              <generateApiDocumentation>false</generateApiDocumentation>

              <modelPackage>com.example.rest</modelPackage>
              <generateModelTests>false</generateModelTests>
              <generateModelDocumentation>false</generateModelDocumentation>

              <configOptions>
                <annotationLibrary>none</annotationLibrary>
                <library>native</library>
                <serializationLibrary>jackson</serializationLibrary>
                <useJakartaEe>true</useJakartaEe>
                <openApiNullable>false</openApiNullable>
                <supportUrlQuery>false</supportUrlQuery>
              </configOptions>
            </configuration>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
</project>
```

##### Related issues/PRs
#17736

##### Suggest a fix
Related pull request

### PR DESCRIPTION
**Pull request by jo1gi**
Fixes #17735

Adds Generated annotations to:
* RFC3339DateFormat
* ServerConfiguration
* ServerVariable
* ApiResponse
* ClassDiscriminatorMapping in JSON.mustache
* TypeReference in api.mustache

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Mentions: @bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608 @martin-mfg

### CODE REVIEW

**Comment by martin-mfg** [comment]
Thanks for the PR!
It mostly looks good to me. But what is the intention behind `GenericTypeRef`?

**Comment by jo1gi** [comment]
> But what is the intention behind `GenericTypeRef`?

The inline `TypeReference` classes are also flagged if they don't have a Generated annotation.

File: samples/client/petstore/kotlin-allOff-discriminator/bin/main/org/openapitools/client/apis/BirdApi.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/**
```
Comment (wing328) [inline samples/client/petstore/kotlin-allOff-discriminator/bin/main/org/openapitools/client/apis/BirdApi.kt:1]: @jo1gi thanks for the PR. where do these new kotlin files come from?

Comment (jo1gi) [reply samples/client/petstore/kotlin-allOff-discriminator/bin/main/org/openapitools/client/apis/BirdApi.kt]: I have no idea. Tried rerunning the sample generation commands.

**Review by martin-mfg — APPROVED** [review]

**Comment by wing328** [comment]
fixed in the latest master but there's another issue with JDK11 as the code now throws exception during runtime.

i've filed https://github.com/OpenAPITools/openapi-generator/pull/18346 to revert back to TypeReference for the time being.

**Pull request merged**